tensorflow/python/eager/def_function.py

```diff
@@ -1192,16 +1192,19 @@ def validate_annotations(self):
     arg_to_dtype = dict(arg_dtype_list)
     type_annotations = self.function_spec.fullargspec.annotations
     for arg, annot in type_annotations.items():
-      if (not hasattr(typing, "_GenericAlias") or
-          not isinstance(annot, typing._GenericAlias)):
+      if not hasattr(annot, '__args__'):
```

Comment:
It's unobvious, so let's add a comment here: "Only verify annotations that are perameterized".

Also, we should check the length, too.